perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/widgets/items/MapBox.java

```diff
@@ -0,0 +1,120 @@
+package cz.metacentrum.perun.wui.registrar.widgets.items;
+
+import com.google.gwt.event.dom.client.BlurEvent;
+import com.google.gwt.event.dom.client.BlurHandler;
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.event.dom.client.ClickHandler;
+import com.google.gwt.user.client.ui.HasHorizontalAlignment;
+import com.google.gwt.user.client.ui.HasVerticalAlignment;
+import com.google.gwt.user.client.ui.HorizontalPanel;
+import com.google.gwt.user.client.ui.Label;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.gwt.user.client.ui.VerticalPanel;
+import cz.metacentrum.perun.wui.widgets.boxes.ExtendedTextBox;
+import cz.metacentrum.perun.wui.model.beans.ApplicationFormItemData;
+import cz.metacentrum.perun.wui.registrar.widgets.PerunForm;
+import cz.metacentrum.perun.wui.widgets.PerunButton;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents MapBox form item.
+ *
+ * @author Jakub Hejda <Jakub.Hejda@cesnet.cz>
+ */
+public class MapBox extends WidgetBox {
+
+	List<ExtendedTextBox> keys;
+	List<ExtendedTextBox> values;
+
+	public MapBox(PerunForm form, ApplicationFormItemData item, String lang) {
+		super(form, item, lang);
+	}
+
+	@Override
+	public String getValue() {
+		String value = "";
+		for (ExtendedTextBox key : keys) {
+			ExtendedTextBox val = values.get(keys.indexOf(key));
+			if (key.getValue() != null && !key.getValue().isEmpty()) {
+				value +=  key.getValue() + ":" + val.getValue() + ",";
+			}
+		}
+
+		return value;
+	}
+
+	@Override
+	protected Widget initWidget() {
+		keys = new ArrayList<>();
+		values = new ArrayList<>();
+		return super.initWidget();
+	}
+
+	@Override
+	protected void setValidationTriggers() {
+		for (ExtendedTextBox input : keys) {
+			input.addBlurHandler(new BlurHandler() {
+				@Override
+				public void onBlur(BlurEvent event) {
+					validateLocal();
+				}
+			});
+		}
+	}
+
+	@Override
+	protected void setValueImpl(String value) {
+		int counter = 0;
+		for (String keyAndValue : value.split(",")) {
+			if (counter != 0) {
+				generateItemWithRemoveButton((VerticalPanel) widget);
+			}
+			String[] keyVal = keyAndValue.split(":");
+			keys.get(counter).setValue(keyVal[0].trim());
+			values.get(counter).setValue(keyVal[1].trim());
+			counter++;
+		}
+	}
+
+	@Override
+	protected PerunButton generateAddButton(VerticalPanel vp) {
+		return new PerunButton("", new ClickHandler() {
+			public void onClick(ClickEvent event) {
+				generateItemWithRemoveButton(vp);
+			}
+		});
+	}
+
+	@Override
+	protected void generateItemWithRemoveButton(VerticalPanel vp) {
+		HorizontalPanel hp = new HorizontalPanel();
+		hp.setVerticalAlignment(HasVerticalAlignment.ALIGN_MIDDLE);
+		hp.setWidth("100%");
+		ExtendedTextBox key = new ExtendedTextBox();
+		key.setPlaceholder(getTranslation().enterKey());
+		Label equals = new Label(" = ");
+		ExtendedTextBox value = new ExtendedTextBox();
```

Comment:
Again we should handle basic regex validation of form item here, I'm just no sure how. Probably should be applied to key=value pairs.